transformer_engine/jax/fp8.py

```diff
@@ -16,10 +16,43 @@
 from transformer_engine.common.recipe import DelayedScaling, Format
 from transformer_engine.jax.sharding import global_shard_guard
 from transformer_engine.jax.sharding import ShardingResource
+from .utils import get_cublasLt_version, get_cuda_version, get_device_compute_capability
 
+_is_fp8_available = None
+_reason_for_no_fp8 = ""
 Collection = Union[Dict, FrozenDict]
 
 
+def _check_fp8_support(gpu_id) -> Tuple[bool, str]:
+    """Return if fp8 support is available"""
+    gpu_arch = get_device_compute_capability(gpu_id)
+    if gpu_arch >= 9.0:    # hopper and above
+        return True, ""
+    if gpu_arch < 8.9:    # pre-ada
+        return False, "Device compute capability 8.9 or higher required for FP8 execution."
+    if get_cublasLt_version() < 120103:
+        return False, "CublasLt version 12.1.3.x or higher required for FP8 execution on Ada."
+    if get_cuda_version() < 12.01:
```

Comment:
Does cuda-python return `12.01` instead of `12.1`?

Reply:
I see that it returns an integer. I think it's better to leave it as an int in that case similar to how to do it for the cublas version (instead of dividing by 1000). So something like `get_cuda_version() < 12010`.

Reply:
Floating-point version numbers are a little suspect:
```
>>> 1.1 == 1.1000000000000001
True
```
I agree leaving it as an `int` would be better. Alternatively, you could break the version up into a tuple of `int` s:
```
>>> (1,1) > (1,0)
True
>>> (1,1) > (1,)
True
>>> (1,1) > (2,0)
False
```

Reply:
Good. I also prefer to keep integater as a version number. The original thinking was to align TE/PyTorch. I will change it later.

Reply:
https://github.com/NVIDIA/TransformerEngine/commit/50d9e4afd2787f45caf22d26eb13fd66ba58fbc3 solved this issue